airflow/jobs/local_task_job.py

```diff
@@ -291,3 +296,13 @@ def _update_dagrun_state_for_paused_dag(self, session=None):
             if dag_run:
                 dag_run.dag = dag
                 dag_run.update_state(session=session, execute_callbacks=True)
+
+    @staticmethod
+    def _enable_task_listeners():
+        """
+        Check if we have any registered listeners, then register sqlalchemy hooks for
+        TI state change if we do.
+        """
+        integrate_listener_plugins()
+        if get_listener_manager().has_listeners():
```

Comment:
`integrate_listener_plugins` should be called automatically from inside `get_listener_manager` I think.

Reply:
Done. Changed `integrate_listener_plugins` to require `ListenerManager` parameter since they would have circular calls otherwise.